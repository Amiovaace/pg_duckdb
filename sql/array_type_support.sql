drop extension if exists quack;
create extension quack;

-- INT4 (single dimension)
CREATE TABLE int_array_1d(a INT[]);
INSERT INTO int_array_1d SELECT CAST(a as INT[]) FROM (VALUES
    ('{1, 2, 3}'),
    ('{4, 5, NULL, 7}'),
    (NULL),
    ('{}')
) t(a);
SELECT * FROM int_array_1d;

-- BOOL (single dimension)
CREATE TABLE bool_array_1d(a BOOL[]);
INSERT INTO bool_array_1d SELECT CAST(a as BOOL[]) FROM (VALUES
    ('{true, false, true}'),
    ('{true, true, NULL, false}'),
    (NULL),
    ('{}')
) t(a);
SELECT * FROM bool_array_1d;

DROP TABLE int_array_1d;
DROP TABLE bool_array_1d;
