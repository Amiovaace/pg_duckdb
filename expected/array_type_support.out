drop extension if exists quack;
create extension quack;
-- INT4 (single dimension)
CREATE TABLE int_array_1d(a INT[]);
INSERT INTO int_array_1d SELECT CAST(a as INT[]) FROM (VALUES
    ('{1, 2, 3}'),
    (NULL),
    ('{4, 5, NULL, 7}'),
    ('{}')
) t(a);
SELECT * FROM int_array_1d;
      a       
--------------
 {1,2,3}
 
 {4,5,NULL,7}
 {}
(4 rows)

-- BOOL (single dimension)
CREATE TABLE bool_array_1d(a BOOL[]);
INSERT INTO bool_array_1d SELECT CAST(a as BOOL[]) FROM (VALUES
    ('{true, false, true}'),
    (NULL),
    ('{true, true, NULL, false}'),
    ('{}')
) t(a);
SELECT * FROM bool_array_1d;
      a       
--------------
 {t,f,t}
 
 {t,t,NULL,f}
 {}
(4 rows)

-- CHAR (single dimension)
CREATE TABLE char_array_1d(a CHAR[]);
INSERT INTO char_array_1d SELECT CAST(a as CHAR[]) FROM (VALUES
    ('{a, b, c}'),
    (NULL),
    ('{t, f, Z, A}'),
    ('{}')
) t(a);
SELECT * FROM char_array_1d;
ERROR:  (DuckDB/ConvertPostgresToDuckColumnType) Unsupported quack type: 1014
DROP TABLE int_array_1d;
DROP TABLE bool_array_1d;
DROP TABLE char_array_1d;
